plugins/imagebase/plugin.js

```diff
@@ -186,7 +216,108 @@
 		};
 	}
 
+	function getCaptionFeature() {
+		function createCaption( widget ) {
+			var element = widget.element,
+				caption = element.getDocument().createElement( 'figcaption' );
+
+			element.append( caption );
+			widget.initEditable( 'caption', widget.definition.editables.caption );
+
+			return caption;
+		}
+
+		function isEmptyOrHasPlaceholder( widget ) {
+			return !widget.editables.caption.getData() || !!widget.parts.caption.data( 'cke-caption-placeholder' );
+		}
+
+		function addPlaceholder( widget ) {
+			widget.parts.caption.data( 'cke-caption-placeholder', widget.editor.lang.imagebase.captionPlaceholder );
+		}
+
+		function removePlaceholder( widget ) {
+			widget.parts.caption.data( 'cke-caption-placeholder', false );
+		}
+
+		function setVisibility( caption, isVisible ) {
+			caption.data( 'cke-caption-active', isVisible );
+			caption.data( 'cke-caption-hidden', !isVisible );
+		}
+
+		return {
+			setUp: function( editor ) {
+				var listeners = [];
+
+				function listener( evt ) {
+					var path = evt.name === 'blur' ? editor.elementPath() : evt.data.path,
+						sender = path ? path.lastElement : null,
+						widgets = editor.widgets,
+						focused = getFocusedWidget( editor ),
+						previous = widgets.getByElement( editor.editable().findOne( 'figcaption[data-cke-caption-active]' ) );
+
+					if ( !editor.filter.check( 'figcaption' ) ) {
+						return CKEDITOR.tools.array.forEach( listeners, function( listener ) {
+							listener.removeListener();
+						} );
+					}
+
+					if ( focused && hasWidgetFeature( focused, 'caption' ) ) {
+						focused._refreshCaption( sender );
+					}
+
+					if ( previous && hasWidgetFeature( previous, 'caption' ) ) {
+						previous._refreshCaption( sender );
+					}
+				}
+
+				listeners.push( editor.on( 'selectionChange', listener , null, null, 9 ) );
+				listeners.push( editor.on( 'blur', listener ) );
+			},
+
+			init: function() {
+				if ( !this.editor.filter.check( 'figcaption' ) ) {
+					return;
+				}
+
+				if ( !this.parts.caption ) {
+					this.parts.caption = createCaption( this );
+				}
+
+				this._refreshCaption();
+			},
+
+			/**
+			 * Method used inside {@link CKEDITOR.editor#event-selectionChange} event to decide if caption for given widget
```

Comment:
You have tied this method strongly with the `selectionChange` and now it uses also `blur`. It's possible there will be different events soon. Please make the docs event-agnostic.